File: leetcode/tree/102.py
# Definition for a binary tree node.
class TreeNode:

    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:

    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        '''队列'''
        if root is None:
            return []
        q = [root]
        ret = []
        while q:
            size = len(q)  #当前层的个数!!!
            t = []
            for _ in range(size):
                node = q.pop(0)
                t.append(node.val)

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            ret.append(t)
        return ret

File: leetcode/tree/102.cpp
#include <iostream>
#include <queue>
#include <vector>

#include "tree_node.h"

class Solution {
 public:
  std::vector<std::vector<int>> levelOrder(TreeNode* root) {
    std::vector<std::vector<int>> vec;
    if (!root) {
      return vec;
    }

    std::queue<TreeNode*> q;
    q.emplace(root);
    while (!q.empty()) {
      size_t level_size = q.size();
      std::vector<int> inner_vec;
      for (size_t i = 0; i < level_size; ++i) {  // 当前层的个数, 遍历当前层
        TreeNode* node = q.front();
        q.pop();
        inner_vec.emplace_back(node->val);

        if (node->left) {
          q.emplace(node->left);
        }
        if (node->right) {
          q.emplace(node->right);
        }
      }

      vec.emplace_back(inner_vec);
    }

    return vec;
  }
};

int main() {
  TreeNode* root = new TreeNode(1);
  root->left = new TreeNode(2);
  root->right = new TreeNode(3);
  root->left->right = new TreeNode(5);
  root->right->left = new TreeNode(6);

  Solution solution = Solution();
  auto ret = solution.levelOrder(root);

  for (const auto& items : ret) {
    for (const auto& item : items) {
      std::cout << item << " ";
    }
    std::cout << std::endl;
  }
  std::cout << std::endl;
}